pkg/backend/cmd/server/server.go

```diff
@@ -628,14 +616,42 @@ func (s *server) Run(ctx context.Context, metrics metrics.Metrics, logger *zap.L
 		pushapi.WithLogger(logger),
 	)
 	pushServer := rpc.NewServer(pushService, *s.certPath, *s.keyPath,
+		"push-server",
 		rpc.WithPort(*s.pushServicePort),
 		rpc.WithVerifier(verifier),
 		rpc.WithMetrics(registerer),
 		rpc.WithLogger(logger),
 	)
-	defer pushServer.Stop(10 * time.Second)
 	go pushServer.Run()
-	// other services...
+	// To detach this pod from Kubernetes Service before the app servers stop, we stop the health check service first.
+	// Then, after 10 seconds of sleep, the application server can be shut down, as no new requests are expected to be sent.
+	// In this case, the Readiness prove must fail within 10 seconds and the pod must be detached.
+	defer func() {
+		go healthcheckServer.Stop(10 * time.Second)
+		time.Sleep(10 * time.Second)
```

Comment:
Since the timeout duration is the same for the other places, how about adding a constant for the timeout?